lib/pf/node.pm

```diff
@@ -1327,6 +1331,32 @@ sub _cleanup_attributes {
     $info->{'status'} = _cleanup_status_value($info->{'status'});
 }
 
+=item node_extended_data
+
+Get the node extended data
+
+=cut
+
+sub node_extended_data {
+    my ($mac) = @_;
+    my @namespaces = qw(mse);
+    if (@{$Config{advanced}{extended_profile_filter_namespaces}}) {
+        my %hash;
+        my $redis = pf::Redis->new(server => 'localhost:6379');
+        my $json = JSON::MaybeXS->new;
+        foreach my $namespace (@namespaces) {
+            my ($data) = $redis->get("extended:${namespace}:${mac}");
```

Comment:
I am not sure how using hashes will prevent race conditions.
Can you please provide a bit more detail on how that will work?

The main reason not use hashes is that we can support different namespaces that each have a  different TTLs.